### PR TITLE
PARQUET-1054: Fixes for Arrow API changes in ARROW-1199

### PR DESCRIPTION
**Pull request by wesm**
This will be broken until ARROW-1199 is merged

### CODE REVIEW

File: src/parquet/arrow/arrow-reader-writer-test.cc [hunk 1]
```diff
@@ -738,7 +740,7 @@ TEST_F(TestUInt32ParquetIO, Parquet_2_0_Compability) {
 TEST_F(TestUInt32ParquetIO, Parquet_1_0_Compability) {
   // This also tests max_definition_level = 1
   std::shared_ptr<Array> arr;
-  ASSERT_OK(NullableArray<::arrow::UInt32Type>(LARGE_SIZE, 100, kDefaultSeed, &arr));
+  ASSERT_OK(NullableArray<::arrow::UInt32Type>(32, 10, kDefaultSeed, &arr));
```
Comment (xhochy) [inline src/parquet/arrow/arrow-reader-writer-test.cc]: Switch back to `LARGE_SIZE`

Comment (wesm) [inline src/parquet/arrow/arrow-reader-writer-test.cc]: done.

**Comment by wesm** [comment]
It looks like MSVC is upset about the conditional compilation of the ZSTD/LZ4 symbols. I will have to fix in Arrow (put those in a separate header file)

**Comment by wesm** [comment]
Appveyor build running here: https://ci.appveyor.com/project/wesm/parquet-cpp/build/1.0.65

**Comment by wesm** [comment]
Appveyor build: https://ci.appveyor.com/project/wesm/parquet-cpp/build/1.0.66

**Pull request closed without merging**